### ISSUE TITLE
entry macros - minimal shorthand entries auto-expanded into full journal entries

### ISSUE THREAD

**Issue by simonmichael** [simonmichael/hledger#259]
I want "entry macros".. where you write just the minimum required info and it gets expanded to a full journal entry at report time (or at data entry time, eg by haskell-mode).

eg: `DATE DESCRIPTION  AMOUNT`
or `DATE DESCRIPTION  ARG1  ARG2  ...`

A predefined entry template is selected by the description.
And possibly if there's no template it'll just use the most recent matching entry as one.

**Comment by achernyak** [comment]
I just started using hledger but had a very similar idea. 

What if we could setup transaction aliases. Say you have a re occurring bill or your oaycheck, and to enter it you can just type in a single alias name. 

This could also meet the above described requirement by letting alias the accounts you want the transection to effect. Then you can simply enter the amount.

**Comment by timbod7** [comment]
I'd like to see macros, but often have more than 2 postings, often with a calculated tax component.

```
2015-04-17  new hard disk
    expenses:equipment            $    159.00
    assets:bank                   $   -159.00
    expenses:tax                  $    -14.45
    liabilities:tax               $     14.45
```

I guess the macro could look something like:

```
entry  equipment AMT
    expenses:equipment            $    AMT
    assets:bank                   $   -AMT
    expenses:tax                  $   -AMT/11
    liabilities:tax               $    AMT/11
```
It would presumably be a fair bit of implementation work to support arithmetic in the macros.

**Comment by simonmichael** [comment]
Indeed. These "macros"/"templates" pop up and overlap in quite a few places:

- shorthand journal entries
- the [add](http://hledger.org/hledger.html#add) command
- converting CSV
- automated transactions/postings (https://github.com/simonmichael/hledger/issues/99)
- [hledger-rewrite](http://hledger.org/hledger.html#rewrite)
